hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/StorageInterfaceImpl.java

```diff
@@ -435,6 +456,24 @@ public void startCopyFromBlob(CloudBlobWrapper sourceBlob, BlobRequestOptions op
           null, dstAccessCondition, options, opContext);
     }
 
+    @Override
+    public void startCopyFromBlob(CloudBlobWrapper sourceBlob, BlobRequestOptions options,
+                                  OperationContext opContext, boolean overwriteDestination, String eTag)
+            throws StorageException, URISyntaxException {
+      AccessCondition dstAccessCondition =
+              overwriteDestination
+                      ? null
+                      : AccessCondition.generateIfNotExistsCondition();
+      if (dstAccessCondition != null) {
+        dstAccessCondition.setIfMatch(eTag);
```

Comment:
the object will have both if-none-match:* and if-match:ETAG.

We should not populate when overwrite condition is on.

Reply:
If-None-Match * is for the PutBlockList temp file flow and If-Match eTag is for the copy blob flow

Reply:
You mean that overwrite will be false in copy flow, and overwrite be true in putblocklist temp file flow?

This is a public method exposed and should be agnostic to what calls it.